packages/feedback/src/integration.ts

```diff
@@ -0,0 +1,460 @@
+          logger.warn(`[Feedback] Unable to find element with selector ${actorTarget}`);
+          return;
+        }
+
+        actorTarget.addEventListener('click', this._handleActorClick);
+      } else {
+        this._createWidgetActor();
+      }
+
+      if (!this._host) {
+        logger.warn('[Feedback] Unable to create host element');
+        return;
+      }
+
+      document.body.appendChild(this._host);
+    } catch (err) {
+      // TODO: error handling
+      console.error(err);
+    }
+  }
+
+  /**
+   * Removes the Feedback widget
+   */
+  public remove(): void {
+    if (this._host) {
+      this._host.remove();
+    }
+  }
+
+  /**
+   * Opens the Feedback dialog form
+   */
+  public openDialog(): void {
+    try {
+      if (this._dialog) {
+        this._dialog.open();
+        this._isDialogOpen = true;
+        if (this.options.onDialogOpen) {
+          this.options.onDialogOpen();
+        }
+        return;
+      }
+
+      try {
+        this._shadow = this._createShadowHost();
+      } catch {
+        return;
+      }
+
+      // Lazy-load until dialog is opened and only inject styles once
+      if (!this._hasDialogEverOpened) {
+        this._shadow.appendChild(createDialogStyles(document));
+      }
+
+      const userKey = this.options.useSentryUser;
+      const scope = getCurrentHub().getScope();
+      const user = scope && scope.getUser();
+
+      this._dialog = Dialog({
+        defaultName: (userKey && user && user[userKey.name]) || '',
+        defaultEmail: (userKey && user && user[userKey.email]) || '',
+        onClosed: () => {
+          this.showActor();
+          this._isDialogOpen = false;
+        },
+        onCancel: () => {
+          this.hideDialog();
+          this.showActor();
+        },
+        onSubmit: this._handleFeedbackSubmit,
+        options: this.options,
+      });
+      this._shadow.appendChild(this._dialog.$el);
+
+      // Hides the default actor whenever dialog is opened
+      this._actor && this._actor.hide();
+
+      this._hasDialogEverOpened = true;
+      if (this.options.onDialogOpen) {
+        this.options.onDialogOpen();
+      }
+    } catch (err) {
+      // TODO: Error handling?
+      console.error(err);
+    }
+  }
+
+  /**
+   * Hides the dialog
+   */
+  public hideDialog = (): void => {
+    if (this._dialog) {
+      this._dialog.close();
+      this._isDialogOpen = false;
+    }
+  };
+
+  /**
+   * Removes the dialog element from DOM
+   */
+  public removeDialog = (): void => {
+    if (this._dialog) {
+      this._dialog.$el.remove();
+      this._dialog = null;
+    }
+  };
+
+  /**
+   * Displays the default actor
+   */
+  public showActor = (): void => {
+    // TODO: Only show default actor
+    if (this._actor) {
+      this._actor.show();
+    }
+  };
+
+  /**
+   * Creates the host element of widget's shadow DOM. Returns null if not supported.
+   */
+  protected _createShadowHost(): ShadowRoot {
+    if (!document.head.attachShadow) {
```

Comment:
I wonder if it wouldn't be easier to not do any check like this, but just try-catch all of this? e.g.:

```ts
protected _createShadowHost(): ShadowRoot {
   // Don't create if it already exists
    if (this._shadow) {
      return this._shadow;
    }

  try {
     // Create the host
    this._host = document.createElement('div');
    this._host.id = this.options.id;

    // Create the shadow root
    const shadow = this._host.attachShadow({ mode: 'open' });

    shadow.appendChild(createMainStyles(document, this.options.colorScheme, this.options.theme));

    return shadow;
  } catch {
     logger.warn('[Feedback] Browser does not support shadow DOM API');
      throw new Error('Browser does not support shadow DOM API.');
  }
}
```

or something like this? May be a bit more robust? and prob. even safes a few bytes..?